erlang_27: 27.0.1 -> 27.0

chiroptical opened the pull request:
Partially reverts NixOS/nixpkgs#326403

The OTP 27.0.1 changes are not building the documentation on Linux. The issue was reported here: https://github.com/NixOS/nixpkgs/pull/326403#issuecomment-2227047895. I wasn't able to reproduce this when building OTP separate from `nix-build -A erlang_27` so I'll need to dig a bit deeper to figure this out.

@happysalada @adamcstephens

chiroptical committed:
erlang_27: 27.0.1 -> 27.0

adamcstephens commented:
Only 27 is broken, I think we probably only want to revert version 27 and not both. It would also be helpful to document why reverting in the commit description.

chiroptical commented:
> Only 27 is broken, I think we probably only want to revert version 27 and not both. It would also be helpful to document why reverting in the commit description.

Fair, I can't handle this now. Ideally tonight.

chiroptical commented:
Okay, I dropped the erlang 25 change, updated to a single commit that I _believe_ follows the contributing guidelines (I've never done a revert). If anyone has any thoughts on why this would fail **only** on Linux I would love to hear it.

chiroptical commented:
> Okay, I dropped the erlang 25 change, updated to a single commit that I _believe_ follows the contributing guidelines (I've never done a revert). If anyone has any thoughts on why this would fail **only** on Linux I would love to hear it.

I was able to make 27.0.1 compile by leaving the glue code in place but I was trying to remove it with PRs submitted to OTP. I'll try to do some more debugging soon.

happysalada commented:
I would say leaving the glue code is fine then (just ny personal opinion, im happy to let you decide what you like best)
Thank you for taking the time to make a PR and investigate !

adamcstephens commented:
I did spend some looking into this last night, but unfortunately made not forward progress. It's almost as if the 27.0.1 build is calling ex_doc incorrectly, but I'd need to spend more time learning the OTP build system to know more.

adamcstephens commented:
Also, thank you @chiroptical and @happysalada . Sorry if I came off harsh in the other PR. I appreciate your contributions.
